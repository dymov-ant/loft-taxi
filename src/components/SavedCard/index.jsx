import React from 'react';
import PropTypes from 'prop-types';
import logoCard from '../../assets/img/icons/card-logo.svg';
import schemaCard from '../../assets/img/icons/card-schema.svg';
import mastercard from '../../assets/img/icons/master-card.svg';
import styles from '../ProfileForm/profileForm.module.scss';

const SavedCard = ({ cardNumber, expiryDate }) => {
  const month = new Date(expiryDate).getMonth() + 1;
  const year = new Date(expiryDate).getFullYear().toString().substr(-2);
  return (
    <div className={styles.card}>
      <div className={styles.card__top}>
        <img className={styles.card__logo} src={logoCard} alt="лого" />
        <p className={styles.card__date}>{`${month}/${year}`}</p>
      </div>
      <p className={styles.card__number}>
        {cardNumber.split(' ').map((block, i) => (
          // eslint-disable-next-line react/no-array-index-key
          <span key={i}>{block}</span>
        ))}
      </p>
      <div className={styles.card__bottom}>
        <img className={styles.card__schema} src={schemaCard} alt="чип" />
        <img className={styles.card__mastercard} src={mastercard} alt="mastercard" />
      </div>
    </div>
  );
};

SavedCard.propTypes = {
  cardNumber: PropTypes.string.isRequired,
  expiryDate: PropTypes.string.isRequired,
};

export default SavedCard;
