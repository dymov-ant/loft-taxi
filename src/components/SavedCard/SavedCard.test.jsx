import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import SavedCard from './index';

describe('SavedCard component', () => {
  const cardNumber = '1111 2222 3333 4444';
  const expiryDate = 'Wed Oct 27 2025 20:04:40 GMT+0500 (Екатеринбург, стандартное время)';
  it('SavedCard renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SavedCard cardNumber={cardNumber} expiryDate={expiryDate} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('SavedCard props', () => {
    render(<SavedCard cardNumber={cardNumber} expiryDate={expiryDate} />);
    expect(screen.getByText('1111')).toBeInTheDocument();
    expect(screen.getByText('10/25')).toBeInTheDocument();
  });

  it('SavedCard snapshot', () => {
    const savedCard = render(<SavedCard cardNumber={cardNumber} expiryDate={expiryDate} />);
    expect(savedCard).toMatchSnapshot();
  });
});
