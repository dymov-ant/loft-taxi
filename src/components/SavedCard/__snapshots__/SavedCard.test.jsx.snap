// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedCard component SavedCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="card__top"
        >
          <img
            alt="лого"
            class="card__logo"
            src="card-logo.svg"
          />
          <p
            class="card__date"
          >
            10/25
          </p>
        </div>
        <p
          class="card__number"
        >
          <span>
            1111
          </span>
          <span>
            2222
          </span>
          <span>
            3333
          </span>
          <span>
            4444
          </span>
        </p>
        <div
          class="card__bottom"
        >
          <img
            alt="чип"
            class="card__schema"
            src="card-schema.svg"
          />
          <img
            alt="mastercard"
            class="card__mastercard"
            src="master-card.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="card__top"
      >
        <img
          alt="лого"
          class="card__logo"
          src="card-logo.svg"
        />
        <p
          class="card__date"
        >
          10/25
        </p>
      </div>
      <p
        class="card__number"
      >
        <span>
          1111
        </span>
        <span>
          2222
        </span>
        <span>
          3333
        </span>
        <span>
          4444
        </span>
      </p>
      <div
        class="card__bottom"
      >
        <img
          alt="чип"
          class="card__schema"
          src="card-schema.svg"
        />
        <img
          alt="mastercard"
          class="card__mastercard"
          src="master-card.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
